cep-20.2.md

```diff
@@ -0,0 +1,831 @@
+  # rpaths_patcher: None
+```
+
+### Script section
+
+```yaml
+script:
+  # the interpreter to use for the script
+  interpreter: str  # defaults to bash on UNIX and cmd.exe on Windows
+  # the script environment. You can use Jinja to pass through environment variables
+  # with the `env` key (`${{ env.get("MYVAR") }}`).
+  env: {str: str}
+  # secrets that are set as env variables but never shown in the logs or the environment
+  secrets: [str]
+  # The file to use as the script. Automatically adds `bat` or `sh` to the filename
+  # on Windows or UNIX respectively (if no file extension is given).
+  file: str  # build.sh  
+  # A string or list of strings that is the script contents (mutually exclusive with `file`)
+  content: str | [str]
+```
+
+## Source section
+
+```yaml
+source: [SourceElement]
+```
+
+where the different source elements are defined as follows.
+
+### URL source
+
+```yaml
+# url pointing to the source tar.gz|zip|tar.bz2|... (this can be a list of mirrors that point to the same file)
+url: url | [url]
+# destination folder in work directory
+target_directory: path
+# rename the downloaded file to this name
+file_name: str
+# hash of the file
+sha256: hex string
+# legacy md5 sum of the file (test both, prefer sha256)
+md5: hex string
+# relative path from recipe file
+patches: [path]
+```
+
+### Local source
+
+A path can be either:
+
+- directory ("../bla")
+- a path to an archive ("../bla.tar.gz")
+- a path to a file ("../bla.txt")
+
+```yaml
+# file, absolute or relative to recipe file
+path: path
+# if there is a gitignore, adhere to it (or not)
+use_gitignore: bool (defaults to true)
+# destination folder
+target_directory: path
+# rename the downloaded file to this name
+file_name: str
+# absolute or relative path from recipe file
+patches: [path]
+```
+
+### Git source
+
+```yaml
+# URL to the git repository or path to local git repository
+git: url | path
+# the following 3 keys are mutually exclusive (branch, tag, and rev)
+# branch to checkout to
+branch: string
+# tag to checkout to
+tag: string
+# revision to checkout to (hash or ref)
+rev: string
+# depth of the git clone (mutually exclusive with rev)
+depth: signed integer (defaults to -1 -> not shallow)
+# should this use git-lfs?
+lfs: bool (defaults to false)
+# destination folder in work directory
+target_directory: path
+# absolute or relative path from recipe file
+patches: [path]
+```
+
+### Removed source definitions
+
+SVN and HG (mercury) source definitions are removed as they are not relevant anymore.
+
+## Requirements section
+
+```yaml
+requirements:
+  # build time dependencies, in the build_platform architecture
+  build: [PackageSelector]
+  # dependencies to link against, in the target_platform architecture
+  host: [PackageSelector]
+  # the section below is copied into the index.json and required at package installation
+  run: [PackageSelector]
+  # constrain optional packages (was `run_constrained`)
+  run_constraints: [PackageSelector]
```

Comment:
Keep in mind that we then have:
- `constrains` as the actual key that is used in the index data later,
- `run_constrained` as the old key in `meta.yaml`,
- `run_constraints` as the new key in `recipe.yaml`.

maybe it makes sense to use a uniform one, i.e., `constrains`(/`run_constrains`)?

(Personally, I don't care much about the naming. But it is bound to confuse people if we use multiple names.)

Reply:
I'll ask this in tonights meeting.

Reply:
Oh, dang, yeah, let's stay consistent

Reply:
https://thecontentauthority.com/blog/constrain-vs-constraint
>‘Constrain’ means to restrict or limit something or someone. It often implies an external force or influence that is imposing the restriction. For example, “The budget constraints prevented us from hiring more staff.”

>‘Constraint’, on the other hand, refers to a limitation or restriction that is inherent in a situation or system. It can also refer to the rules or regulations that govern a particular activity or process. For example, “The constraints of the project timeline meant that we had to work overtime to meet the deadline.”

Here we really mean constraints right?

Reply:
The example for `constrains` uses `constraints`? :D 

I think one can say that these are "added constraints" for the environment. But I am not a native english speaker which is partially why I want to ask in 4 minutes :)

Reply:
The index name is wrong according to my native ears but matches "depends" (which is also not a noun). "Depends" also does not match its corresponding section in requirements/run which is another fun mismatch.

Reply:
The index is really an internal api and so I don't think should influence the recipe spec which is meant for use by folks not developing conda build.

Reply:
Hence I agree with constraints